daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -120,7 +122,11 @@ private[lf] object Speedy {
       var globalDiscriminators: Set[crypto.Hash],
       var cachedContracts: Map[V.ContractId, CachedContract],
   ) extends LedgerMode {
-
+    private[lf] val visibleToStakeholders: Set[Party] => SVisibleToStakeholders =
+      if (validating) { _ => SVisibleToStakeholders.Visible }
```

Comment:
Naive question, why do we skip the check on validation ?

Reply:
Visibility checks are a submission-only concept. They exist primarily to protect parties on the same participant from each other.